repos/system_upgrade/el7toel8/actors/checkkdegnome/actor.py

```diff
@@ -0,0 +1,137 @@
+        self.log.info("----------------------------------")
+
+        # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+        if not kde_desktop_installed and not gnome_desktop_installed:
+            self.log.info("INFO: No desktop installed. Continuing with the upgrade.")
+            self.produce(CanUpgrade(canUpgrade=0))
+            return
+
+        if kde_desktop_installed:
+            self.log.info("INFO: KDE desktop is installed. Checking what we can do about it.")
+            if not gnome_desktop_installed:
+                self.log.info("ERROR: Cannot perform the upgrade because there is no other desktop installed.")
+                self.produce(CanUpgrade(canUpgrade=1))
+                return
+
+            # Assume GNOME is installed in this state
+
+            user = os.environ.get("USER")
+            default_xsession = self.get_xsession("/var/lib/AccountsService/users/{0}".format(user))
+            if not default_xsession:
+                self.log.info("WARN: Unable to detect default session.")
+            else:
+                if "plasma" in default_xsession:  # using in because there can be some white spaces
+                    self.log.info("INFO: KDE used as default session.")
+                    self.log.info("INFO: Upgrade can be performed, but KDE desktop will be removed in favor of GNOME")
+                    self.produce(CanUpgrade(canUpgrade=2))
+                    return  # upgrade can be performed, but user will loose KDE desktop in favor of GNOME desktop
+                else:
+                    self.log.info("INFO: GNOME used as default session. Continuing with the upgrade.")
+            self.log.info("----------------------------------")
+
+        # At this state we can assume that KDE desktop as such is not installed or used and we just need to
+        # detect whether any KDE/Qt app is actively used to inform user that the application will be removed
+        # during the upgrade process
+
+        base_kde_apps = ("kde-baseapps",
+                         "okular",
+                         "ark",
+                         "kdepim",
+                         "konsole",
+                         "gwenview",
+                         "kdenetwork",
+                         "kate", "kwrite")
+        self.log.info("  Detecting installed KDE apps  ")
+        self.log.info("================================")
+
+        for app in base_kde_apps:
+            if self.is_installed(app) == _ret_OK:  # return code of "rpm -q app" command
+                self.log.info("* {0} is installed: True".format(app))
+                self.check_app_in_use(app)
+            else:
+                self.log.info("* {0} is installed: False".format(app))
+
+        self.log.info("----------------------------------")
+
+        if apps_in_use > 0:
+            self.log.info("INFO: KDE apps in use detected.")
```

Comment:
No need to prepend the log message with the log level like "INFO: ". That's being added by the logging handler automatically.